File: srna/reads_length_dist.py
import os
## from glob import glob
import pandas as pd

# fs = glob("blastn/*.align.tsv") + glob("clean/*.tsv")
fs = os.sys.argv[1:]
for f in fs:
    if f.endswith(".length.tsv"): continue
    d1 = pd.read_csv(f, sep="\t")[["length", "copy"]]
    s1 = d1.groupby(["length"])["copy"].sum()
    s2 = pd.DataFrame({"count":s1.values}, index=s1.index)
    s2.to_csv(f.replace(".tsv", ".length.tsv"), sep="\t")

File: srna/reads_length_dist.r
library(ggplot2)

# fs <- c(Sys.glob("blastn/*.length.tsv"), Sys.glob("clean/*.length.tsv"))
Args <- commandArgs(TRUE)
fs = Args

for (f in fs) {
  d <- read.delim(f)
  PDF <- sub(".tsv", ".pdf", f)

  p <- ggplot(d, aes(x=length, y=count)) + 
    geom_bar(stat="identity", fill="#4682B4") +
    ## theme_bw() +
    theme(plot.title = element_text(hjust = 0.5) ) +
    ggtitle("Length distribution") + 
    ylab("") + xlab("") + scale_x_continuous(breaks = seq(min(d$length), max(d$length), by = 2)) +
    geom_text(aes(label=d$count), vjust=-0.5, size=2)

  ggsave(PDF, p)
}
